### PR TITLE
feat(backend): seed posts as article

### PR DESCRIPTION
**Pull request by Mogge**
### Todo
- [x] After #6008 is merged,fix failing end-to-end tests

### CODE REVIEW

**Review by Elweyn — APPROVED** [review]
I know it is breaking the end-to-end test but I don't understand why, so I would like to merge it for now and fix the last things on the event-master branch

**Comment by mahula** [comment]
@Mogge can we have-a-have-a look at the failing [e2e scenario (Moderation.ReportContent - Review reported content)](https://github.com/Ocelot-Social-Community/Ocelot-Social/actions/runs/5092481989)?

My apporach and findings so far:

- (try to) reproduce the test flow behavior on Guthub Actions CI
  - locally checkout branch `seed-posts-as-articles`
  - run as in the [e2e workflow file](https://github.com/Ocelot-Social-Community/Ocelot-Social/blob/14ed84620ba811d8d490906f61c9c81a43250da4/.github/workflows/test-e2e.yml)
    - `compose -f docker-compose.yml -f docker-compose.test.yml up --detach --no-deps webapp neo4j backend`
    - `yarn`
    - `yarn run cypress:run --spec cypress/integration/Moderation.ReportContent.feature`
- add explicit waiting and check of correct report seeding response at precondition [Given somebody reported the following posts](https://github.com/Ocelot-Social-Community/Ocelot-Social/blob/14ed84620ba811d8d490906f61c9c81a43250da4/cypress/integration/Moderation.ReportContent/somebody_reported_the_following_posts.js)

- the error is locally hardly reproducable on contemporary equipped computers
  - the [Github runnner specs](https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners) say
    - 2-core CPU (x86_64)
    - 7 GB of RAM
    - 14 GB of SSD space
  - on a machine relatively meeting the Github runners specs (2 cores, 10 GB RAM) the error is reproducable
    - even when waiting for about 10 minutes at the following two steps the result at the moderator's page is the empty  "no reports"
      - after the seeding step `Given somebody reported the following posts`
      - after clicking the avatar menu "moderation" in `And I click on "Moderation"`

**Comment by mahula** [comment]
@Mogge can we have-a-have-a look at the failing [e2e scenario (Moderation.ReportContent - Review reported content)](https://github.com/Ocelot-Social-Community/Ocelot-Social/actions/runs/5092481989)?

My apporach and findings so far:

#### Reproduce the Guthub test flow context and behavior locally

##### Run tests locally as specified in end-to-end test workflow
- checkout branch `seed-posts-as-articles`
- run as in the [e2e workflow file](https://github.com/Ocelot-Social-Community/Ocelot-Social/blob/14ed84620ba811d8d490906f61c9c81a43250da4/.github/workflows/test-e2e.yml)
  - `compose -f docker-compose.yml -f docker-compose.test.yml up --detach --no-deps webapp neo4j backend`
  - `yarn`
  - `yarn run cypress:run --spec cypress/integration/Moderation.ReportContent.feature`

- adapt to Github hosted runners' specs
  - the e2e test failing is hardly reproducable on contemporary well equipped machines
  - :information_source: on a 2-core machine relatively meeting the [Github runnner specs](https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners) the error is reproducable (no reports on the moderator's page)
6227-no_reports

#### Reason behind the empty reports list on less powerful machines

##### Add network interception to tests for decent request result checks
- at report seeding request in preparation step [Given somebody reported the following posts](https://github.com/Ocelot-Social-Community/Ocelot-Social/blob/14ed84620ba811d8d490906f61c9c81a43250da4/cypress/integration/Moderation.ReportContent/somebody_reported_the_following_posts.js)
- at step [I see all the reported posts including the one from above](https://github.com/Ocelot-Social-Community/Ocelot-Social/blob/eecdb66024a6dce2e246ea89bef01549cceb8fa8/cypress/integration/Moderation.ReportContent/I_see_all_the_reported_posts_including_the_one_from_above.js)

##### 
On the 2-core machine a GraphQLError is revealed

https://github.com/Ocelot-Social-Community/Ocelot-Social/assets/3883288/2a241572-94ee-44f4-bf03-d0941d454afa

###### Stacktrace
```
0: "GraphQLError: Abstract type ReportedResource must resolve to an Object type at runtime for field
    Report.resource with value { post: null, author: { encryptedPassword:
    "$2a$10$NBmGNAFun0dhkeHz93RIi.BhNCfZx.tq1fr4/1FOhP0Eis6ZGNloW", name: "I'm gonna mute Moderators and
    Admins HA HA HA", updatedAt: "2023-05-28T08:30:31.184Z", role: "user", termsAndConditionsAgreedAt:
    "2019-08-01T10:47:19.212Z", slug: "annoying", locale: "en", sendNotificationEmails: true, id:
    "annoying-user", termsAndConditionsAgreedVersion: "0.0.4", createdAt: "2023-05-28T08:30:31.184Z",
    disabled: false, allowEmbedIframes: false, showShoutsPublicly: false, deleted: false }, __typename: "Article", 
    updatedAt: "2023-05-28T08:30:31.423Z", clickedCount: { low: 0, high: 0 }, slug: "the-truth-about-the-holocaust",
    viewedTeaserCount: { low: 1, high: 0 }, id: "p1", language: "en", contentExcerpt: "It never existed!",
    createdAt: "2023-05-28T08:30:31.423Z", content: "It never existed!", disabled: false, title: "The Truth
    about the Holocaust", deleted: false }, received "undefined". Either the ReportedResource type should
    provide a "resolveType" function or each possible type should provide an "isTypeOf" function."

1: "    at ensureValidRuntimeType (/app/node_modules/graphql/execution/execute.js:667:11)"
2: "    at completeAbstractValue (/app/node_modules/graphql/execution/execute.js:660:42)"
3: "    at completeValue (/app/node_modules/graphql/execution/execute.js:585:12)"
4: "    at completeValue (/app/node_modules/graphql/execution/execute.js:557:21)"
5: "    at /app/node_modules/graphql/execution/execute.js:492:16"
6: "    at processTicksAndRejections (node:internal/process/task_queues:95:5)"
7: "    at async Promise.all (index 5)"
8: "    at async Promise.all (index 0)"
9: "    at async Promise.all (index 0)"
```

File: backend/src/db/factories.js [hunk 1]
```diff
@@ -165,7 +165,7 @@ Factory.define('post')
   })
   .after(async (buildObject, options) => {
     const [post, author, image, /* categories, */ tags] = await Promise.all([
-      neode.create('Post', buildObject),
```
Comment (ulfgebhardt) [inline backend/src/db/factories.js:168]: it must be both - Article and Post

Comment (Mogge) [reply backend/src/db/factories.js:168]: No, no, get it somehow working.
When we check out the code, seed and query the data, everything works as expected. Later, using the API on the seeded data, everything works as expected. The only thing that fails, is the end to end test. @ulfgebhardt

File: cypress/support/step_definitions/Moderation.ReportContent/I_see_all_the_reported_posts_including_the_one_from_above.js [hunk 2]
```diff
@@ -1,6 +1,108 @@
...
+      }
+      createdAt
+      reasonCategory
+      reasonDescription
+      __typename
+    }
+    __typename
+  }
+}
+`
```
Comment (Elweyn) [inline cypress/support/step_definitions/Moderation.ReportContent/I_see_all_the_reported_posts_including_the_one_from_above.js:99]: Wenn ich richtig pingelig sein wollte würde ich hierzu sagen könnte nochmal schöner eingerückt werden...
Aber dazu kommt ja bald der linter!

**Review by Elweyn — APPROVED** [review]
Tests are working! Nice work!

**Pull request merged**